pkg/proxy/redisconn/pool.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2015 Reborndb Org. All Rights Reserved.
+// Licensed under the MIT (MIT-LICENSE.txt) license.
+
+package redisconn
+
+import (
+	"sync"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/pools"
+)
+
+type CreateConnFunc func(addr string) (*Conn, error)
+
+type Pool struct {
+	p *pools.ResourcePool
+}
+
+func NewPool(addr string, capability int, f CreateConnFunc) *Pool {
+	poolFunc := func() (pools.Resource, error) {
+		return f(addr)
+	}
+
+	p := new(Pool)
+	p.p = pools.NewResourcePool(poolFunc, capability, capability, 120*time.Second)
+	return p
+}
+
+func (p *Pool) GetConn() (*Conn, error) {
+	conn, err := p.p.Get()
+	if err != nil {
```

Comment:
can we use switch instead. it is preferred in Go than using if ... else if ... else if